include/libopencm3/stm32/common/dsi_common_f47.h

```diff
@@ -44,779 +44,1667 @@
+name     | bits  | description              | default
+-------- | ----- | ------------------------ | -------
+Reserved | 31:14 |                          | 0
+VA       | 13:0  | Vertical active duration | 0
+@{*/
+#define DSI_VVACR_VA_MASK              0x00003FFF
+#define DSI_VVACR_VA_SHIFT             0
+/*@}*/
+
+/** @defgroup dsi_lccr_values DSI_LCCR values
+ * @ingroup dsihost_registers
+ * @brief  DSI Host LTDC command configuration
+name     | bits  | description  | default
+-------- | ----- | ------------ | -------
+Reserved | 31:16 |              | 0
+CMDSIZE  | 15:0  | Command size | 0
+@{*/
+#define DSI_LCCR_CMDSIZE_MASK          0x0000FFFF
+#define DSI_LCCR_CMDSIZE_SHIFT         0
+/*@}*/
+
+/** @defgroup dsi_cmcr_values DSI_CMCR values
+ * @ingroup dsihost_registers
+ * @brief  DSI Host command mode configuration
+name     | bits  | description                                      | default
+-------- | ----- | ------------------------------------------------ | -------
+Reserved | 31:25 |                                                  | 0
+MRDPS    | 24    | Maximum read packet size                         | 0
+Reserved | 23:20 |                                                  | 0
+DLWTX    | 19    | DCS long write transmission                      | 0
+DSR0TX   | 18    | DCS short read zero parameter transmission       | 0
+DSW1TX   | 17    | DCS short read one parameter transmission        | 0
+DSW0TX   | 16    | DCS short write zero parameter transmission      | 0
+Reserved | 15    |                                                  | 0
+GLWTX    | 14    | Generic long write transmission                  | 0
+GSR2TX   | 13    | Generic short read two parameters transmission   | 0
+GSR1TX   | 12    | Generic short read one parameters transmission   | 0
+GSR0TX   | 11    | Generic short read zero parameters transmission  | 0
+GSW2TX   | 10    | Generic short write two parameters transmission  | 0
+GSW1TX   | 9     | Generic short write one parameters transmission  | 0
+GSW0TX   | 8     | Generic short write zero parameters transmission | 0
+Reserved | 7:2   |                                                  | 0
+ARE      | 1     | Acknowledge request enable                       | 0
+TEARE    | 0     | Tearing effect acknowledge request enable        | 0
+@{*/
+typedef enum {
+	DSI_CMCR_MRDPS                   = 1<<24,
+	DSI_CMCR_DLWTX                   = 1<<19,
+	DSI_CMCR_DSR0TX                  = 1<<18,
+	DSI_CMCR_DSW1TX                  = 1<<17,
+	DSI_CMCR_DSW0TX                  = 1<<16,
+	DSI_CMCR_GLWTX                   = 1<<14,
+	DSI_CMCR_GSR2TX                  = 1<<13,
+	DSI_CMCR_GSR1TX                  = 1<<12,
+	DSI_CMCR_GSR0TX                  = 1<<11,
+	DSI_CMCR_GSW2TX                  = 1<<10,
+	DSI_CMCR_GSW1TX                  = 1<<9,
+	DSI_CMCR_GSW0TX                  = 1<<8,
+	DSI_CMCR_ARE                     = 1<<1,
+	DSI_CMCR_TEARE                   = 1<<0
+} dsi_cmcr_flags_t;
```

Comment:
I'm not _at all_ a fan of trying this alternative style of defining all the fields. it's not done anywhere else, and it adds lots of typedefs.  What's the motivation for going for a completely new style for bit definitions?